### PR TITLE
Convert deploy versions to ints when comparing

### PR DESCRIPTION
**Pull request by zpatrick**
Fixes https://github.com/quintilesims/layer0/issues/27

### CODE REVIEW

File: api/handlers/tag_handler_test.go [hunk 1]
```diff
@@ -173,7 +182,51 @@ func TestFindTags_byVersion(t *testing.T) {
...
-				r.AssertEqual(tags[0].EntityID, "d2")
+				r.AssertEqual(tags[0].EntityID, "d.2")
+			},
+		},
+		{
+			Name: "7",
+			Request: &TestRequest{
+				Query: "version=7",
+			},
+			Run: func(r *testutils.Reporter, _ interface{}, req *restful.Request, resp *restful.Response, read Readf) {
```
Comment (tlake) [inline api/handlers/tag_handler_test.go]: [nit] Maybe these tests could be further parametrized? Programmatically generate both the `[]*models.Tag{}`s on line 155 and the `[]HandlerTestCase{}`s here? Excepting the last test case here where we're making sure that 84 doesn't exist, only a few numbers in some strings are changing between tests.

File: common/testutils/reporter.go [hunk 2]
```diff
@@ -18,49 +18,49 @@ func NewReporter(t *testing.T, name string) *Reporter {
 	}
 }
 
-func (this *Reporter) Error(err error) {
-	namedFormat := this.formatName(err.Error())
-	this.T.Errorf(namedFormat)
+func (r *Reporter) Error(err error) {
```
Comment (tlake) [inline common/testutils/reporter.go]: Removing more `this`s be like:

File: api/handlers/tag_handler_test.go (hunk 1 above)
Comment (zpatrick) [inline api/handlers/tag_handler_test.go]: :+1: good call

Comment (zpatrick) [inline api/handlers/tag_handler_test.go]: I've added the `callAndRead` function to remove some of the duplicate code in these cases.

**Comment by tlake** [comment]
changes look good

**Pull request merged**